compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java

```diff
@@ -144,7 +144,10 @@ private void performCodeGen() {
 
         List<Diagnostic> diagnostics = new ArrayList<>();
         for (ModuleContext moduleContext : pkgResolution.topologicallySortedModuleList()) {
-            moduleContext.generatePlatformSpecificCode(compilerContext, this);
+            // We can't generate backend code when one of its dependencies have errors.
```

Comment:
Will this interop validation errors? We already have a check before codegen after interop validation. https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java#L275

Reply:
If there are errors in dependencies they yes, this will not goto interpo validation

Reply:
I think this is safe for now. We have this problem in General. Let's figure out a proper solution later. wdyt ?

Reply:
+1. BTW we can move the      if (diagnostics.isEmpty()) { check after line 151
 moduleContext.diagnostics().forEach(diagnostic ->
                    diagnostics.add(new PackageDiagnostic(diagnostic, moduleContext.moduleName())));

AFAIU first check will be always true since diagnostics isEmpty.

Reply:
With fix for this comment https://github.com/ballerina-platform/ballerina-lang/pull/30234/files#r622735337 we can keep this as is right